src/editor/css_flatten.rs

```diff
@@ -0,0 +1,91 @@
+use scraper::Selector;
+
+pub fn apply(selector: &Selector, html: &str) -> String {
+    let mut html = scraper::Html::parse_document(html);
+    let selected = html
+        .select(selector)
+        .map(|element| element.id())
+        .collect::<Vec<_>>();
+    for id in selected {
+        let children = html
+            .tree
+            .get(id)
+            .map(|parent| {
+                parent
+                    .children()
+                    .map(|child| child.id())
+                    .collect::<Vec<_>>()
+            })
+            .unwrap_or_default();
+        if let Some(mut node_mut) = html.tree.get_mut(id) {
+            for id in children {
+                node_mut.insert_id_before(id);
+            }
+            node_mut.detach();
+        }
+    }
+    html.html()
+}
+
+/// Test [`apply`] with a simplified selector input.
+#[cfg(test)]
+#[track_caller]
+fn ta(selector: &str, html: &str) -> String {
+    let selector = Selector::parse(selector).unwrap();
+    apply(&selector, html)
+}
+
+#[cfg(test)]
+const EXAMPLE_HTML: &str =
+    r#"<html><head></head><body><div class="a"><p>A</p></div><div class="b">B</div></body></html>"#;
+
+#[test]
+fn nothing_selected_changes_nothing() {
+    assert_eq!(ta("span", EXAMPLE_HTML), EXAMPLE_HTML);
+}
```

Comment:
```suggestion
fn select(selector: &str) -> Selector {
    let selector = Selector::parse(selector).unwrap()
}

#[cfg(test)]
const EXAMPLE_HTML: &str =
    r#"<html><head></head><body><div class="a"><p>A</p></div><div class="b">B</div></body></html>"#;

#[test]
fn nothing_selected_changes_nothing() {
    assert_eq!(apply(select("span"), EXAMPLE_HTML), EXAMPLE_HTML);
}
```
Hm just a thought, might make the test more readable / comprehensible?

Reply:
Moved the `assert_eq` into the case method. Maybe it's a bit more clear now?

Reply:
Hm, I would say it remains just as clear as my suggestion. Reason for that is, that your new code is "style wise" or "rust wise" better than my suggestion. But it introduces the downside that the unit test no longer look like an example on how to use this editor (programming API wise).

Since you repeatedly told me: Teufelchen, this is not a library but an end-user program - Your new code should be better (I just have to look away 😶‍🌫️😶‍🌫️😶‍🌫️)

Reply:
I thought about having a general selector method / macro somewhere which is just imported but that diverges even more from the real world usage then as its even further away from the actual usage and less transparent what happens then. So… I liked exploring the idea again, but it seems less optimal than I hoped for at the beginning.

Also, I think in the Rust world tests shouldn't have the goal of being like they are used by the developer using a library. That's what doc tests are for: documenting the usage while giving an example how to use it at the same time. Tests should only help to find implementation bugs in a simple to do way.